### PR TITLE
[API / Admin] BOOK-13/14/31 - Implement Reviews Endpoints and Views 

### PR DESCRIPTION
**Pull request by Salsa-beel**

### CODE REVIEW

**Review by YoussefWaelMohamedLotfy — CHANGES_REQUESTED** [review]
@Salsa-beel You have forgot to add the *Paginated* Endpoints to your Controllers. Kindly add them to proceed with the merge.

**Comment by Salsa-beel** [comment]
pagination part is done.
mappings are added too.

**Comment by YoussefWaelMohamedLotfy** [comment]
> pagination part is done. mappings are added too.

@Salsa-beel Nothing has been pushed

**Comment by Salsa-beel** [comment]
Can you check again now ?

**Comment by YoussefWaelMohamedLotfy** [comment]
image
@Salsa-beel Still nothing. When you find a commit tag like this in this PR discussion, it means it is available to public

**Comment by Salsa-beel** [comment]
Screenshot 2023-04-04 at 9 16 07 PM

but I have committed and pushed all the changes and here is the history of the branch

**Comment by YoussefWaelMohamedLotfy** [comment]
@Salsa-beel It turns out that you are pushing to a different Branch, named `Book-13-Implement-Hotel-Reviews` while this PR is for `dev` 😐

File: src/Services/BookingClone.API/Controllers/V1/AttractionReviewsController.cs
Comment (YoussefWaelMohamedLotfy) [inline src/Services/BookingClone.API/Controllers/V1/AttractionReviewsController.cs:1]: Missing Endpoints for pagination

File: src/Services/BookingClone.API/Controllers/V1/HotelReviewsController.cs
Comment (YoussefWaelMohamedLotfy) [inline src/Services/BookingClone.API/Controllers/V1/HotelReviewsController.cs:1]: Missing Endpoints for pagination

File: src/Services/BookingClone.Application/Features/AttractionReviewFeatures/EntityMappingConfig.cs
Comment (YoussefWaelMohamedLotfy) [inline src/Services/BookingClone.Application/Features/AttractionReviewFeatures/EntityMappingConfig.cs:1]: What about the actual Mapping? 🤣

File: src/Services/BookingClone.Application/Features/HotelReviewFeatures/EntityMappingConfig.cs
Comment (YoussefWaelMohamedLotfy) [inline src/Services/BookingClone.Application/Features/HotelReviewFeatures/EntityMappingConfig.cs:1]: What about the actual Mapping? 🤣

**Review by YoussefWaelMohamedLotfy — CHANGES_REQUESTED** [review]
Please complete the mentioned

**Comment by Salsa-beel** [comment]
@YoussefWaelMohamedLotfy 

Screenshot 2023-04-04 at 9 16 07 PM

aren't these the pagination part and entity map config file ? or you meant something else ?

**Comment by YoussefWaelMohamedLotfy** [comment]
> aren't these the pagination part and entity map config file ? or you meant something else ?

@Salsa-beel Your Query Handlers are done, but throw exceptions when trying to invoke them because of not configuring mapping. You added files for configuring the mappings but did not add the maps 😞 Also, you should an endpoint for Pagination in the controllers. Kindly finish and test then ask for a review again when done.

File: src/WebApps/BookingClone.Admin/EntityMappingConfig.cs
Comment (YoussefWaelMohamedLotfy) [inline src/WebApps/BookingClone.Admin/EntityMappingConfig.cs:1]: Correct Configs, but wrong place 👀

Comment (Salsa-beel) [reply src/WebApps/BookingClone.Admin/EntityMappingConfig.cs:1]: did all the changes needed, added the maps in the entity map config and made sure the file is in the correct place, deleted the wrong one, also added the pagination code needed in the hotel review & attraction review controller, all these changes are done on the dev branch, hopefully this time it works out, let me know if anything else needs to be edited.

File: src/Services/BookingClone.Application/Features/AttractionReviewFeatures/Queries/GetAllAttractionReviews/GetAllAttractionReviewsQueryHandler.cs
Comment (YoussefWaelMohamedLotfy) [inline src/Services/BookingClone.Application/Features/AttractionReviewFeatures/Queries/GetAllAttractionReviews/GetAllAttractionReviewsQueryHandler.cs:1]: @Salsa-beel Missing Logic here

Comment (Salsa-beel) [inline src/Services/BookingClone.Application/Features/AttractionReviewFeatures/Queries/GetAllAttractionReviews/GetAllAttractionReviewsQueryHandler.cs:1]: done

**Review by YoussefWaelMohamedLotfy — APPROVED** [review]

**Pull request merged**